### PR TITLE
Streamline and rearrange Java tests

### PR DESCRIPTION
**Pull request by benwarfield-usds**
## Related Issue or Background Info

- our API tests are a little entangled with too many things in the same package
- our test resources root folder is waaay overcrowded
- we have some tests that do more work than they need to

## Changes Proposed

- moved all GraphQL queries and tests down a level in the directory hierarchy
- changed some tests from full-integration tests to lighter-weight tests
- did some other test clean-up while I was in the neighborhood

## Additional Information

Most of this is developer convenience stuff, but some of the tests will run a lot faster when run individually now.

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/service/email/EmailService.java [hunk 1]
```diff
@@ -7,13 +7,17 @@
...
 import org.springframework.stereotype.Service;
 
 @Service
 public class EmailService {
-  @Autowired private EmailProvider emailProvider;
-  @Autowired private SendGridProperties sendGridProperties;
+  private final EmailProvider emailProvider;
+  private final SendGridProperties sendGridProperties;
+
+  public EmailService(EmailProvider emailProvider, SendGridProperties sendGridProperties) {
```
Comment (nullflux) [inline backend/src/main/java/gov/cdc/usds/simplereport/service/email/EmailService.java:17]: Good call ✅

File: backend/src/test/java/gov/cdc/usds/simplereport/service/email/EmailServiceTest.java [hunk 2]
```diff
@@ -5,23 +5,36 @@
...
+      new SendGridProperties(true, null, "me@example.com", List.of(), List.of());
+
+  @Mock EmailProvider mockSendGrid;
   @Captor ArgumentCaptor<Mail> mail;
 
+  private EmailService _service;
+
+  @BeforeEach
+  void initService() {
+    _service = new EmailService(mockSendGrid, FAKE_PROPERTIES);
```
Comment (nullflux) [inline backend/src/test/java/gov/cdc/usds/simplereport/service/email/EmailServiceTest.java:32]: 👍

**Review by nullflux — APPROVED** [review]
LGTM

**Review by jeskew-gov — APPROVED** [review]

**Review by ClydeDroid — APPROVED** [review]
Thanks for this!

**Pull request merged**